### PR TITLE
Source reference for pattern

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by leonardoalt** [comment]
is this reviewable>

**Comment by chriseth** [comment]
Yes, starting from https://github.com/powdr-labs/powdr/pull/1411/commits/5ea4ac7aa5af96e00de9359ac37293851fd77284

File: importer/src/path_canonicalizer.rs [hunk 1]
```diff
@@ -684,7 +680,7 @@ fn check_expression(
             check_expressions(location, items, state, local_variables)
         }
         Expression::LambdaExpression(
-            source_ref,
+            _source_ref,
```
Comment (Schaeff) [inline importer/src/path_canonicalizer.rs:683]:
```suggestion
            _,
```

File: pil-analyzer/src/type_inference.rs [hunk 2]
```diff
@@ -538,14 +538,12 @@ impl<'a> TypeChecker<'a> {
                     .map(|item| self.infer_type_of_expression(item))
                     .collect::<Result<_, _>>()?,
             }),
-            Expression::LambdaExpression(source_ref, LambdaExpression { kind, params, body }) => {
+            Expression::LambdaExpression(_source_ref, LambdaExpression { kind, params, body }) => {
```
Comment (Schaeff) [inline pil-analyzer/src/type_inference.rs:541]:
```suggestion
            Expression::LambdaExpression(_, LambdaExpression { kind, params, body }) => {
```

**Review by Schaeff — APPROVED** [review]

**Review by Schaeff — APPROVED** [review]

**Pull request merged**